src/helpers/helpers.ts

```diff
@@ -111,6 +112,50 @@ const normalizeHosts = (data?: string) =>
     ?.split(',')
     ?.filter(Boolean);
 
+/**
+ * Validates hosts textarea content.
+ *
+ * @param {string} data - host textarea content.
+ * @param {number} maxHosts - maximum number of hosts that the field should allow.
+ * @returns ValidatedOptions
+ */
+const validateHosts = (data: string | undefined, maxHosts: number) => {
```

Comment:
required: lets use a default for `maxHosts`. reason... old skool js, secondary params are associated with support settings and we can provide a tiny bit of developer "ease of use"

up to you whether you want to use `maxHosts: number = Infinity` or `maxHosts: number = 1` (i partially assume it would default to Infinity)

once you do this your typescript can also be updated towards (and of course the call where you declare `Infinity` in `addSourceModal` can be updated)

```
const validateHosts = (data?: string, maxHosts: number = Infinity) => {
```

Reply:
Done c8a96d131f94ebd901a405ef37d2ca19ebca73a5 .

I still supply both arguments on caller side, to make it more explicit that there are different rules based on source type.